### PR TITLE
fix(migrate): load .env file for migrate diff and db execute

### PR DESCRIPTION
**Pull request by Jolg42**
Previous PR https://github.com/prisma/prisma/pull/11521 removed global dotenv loading didn't add it for `migrate diff` & `db execute` where it's actually needed 

`migrate diff` needs it for `--from-schema-datasource` and `--to-schema-datasource`
`db execute` needs it for `--schema`

### CODE REVIEW

**Review by tomhoule — APPROVED** [review]
looks like that should fix it :+1:

File: packages/migrate/src/commands/MigrateDiff.ts [hunk 1]
```diff
@@ -147,6 +147,8 @@ ${chalk.bold('Examples')}
       throw new MigrateDiffNeedsPreviewFeatureFlagError()
     }
 
+    loadEnvFile(args['--schema'], false)
```
Comment (tomhoule) [inline packages/migrate/src/commands/MigrateDiff.ts:150]: there is no `--schema` argument to diff, is this line right?

File: packages/migrate/src/commands/DbExecute.ts [hunk 2]
```diff
@@ -91,6 +91,8 @@ ${chalk.bold('Examples')}
       throw new DbExecuteNeedsPreviewFeatureFlagError()
     }
 
+    loadEnvFile(args['--schema'], false)
```
Comment (tomhoule) [inline packages/migrate/src/commands/DbExecute.ts:94]: I think same as previous comment

File: packages/migrate/src/commands/MigrateDiff.ts (hunk 1 above)
Comment (Jolg42) [reply packages/migrate/src/commands/MigrateDiff.ts]: Indeed! A bit surprised the type safety didn't help...

**Pull request merged**